coding/src/reed_solomon/mod.rs

```diff
@@ -419,58 +429,82 @@ fn decode<H: Hasher>(
 ///    generated. This new root MUST match the original [bmt] root. This prevents attacks where
 ///    an adversary provides a valid set of chunks that decode to different data.
 /// 4. If the roots match, the original data is extracted from the reconstructed data shards.
+#[derive(Clone, Copy)]
 pub struct ReedSolomon<H> {
     _marker: PhantomData<H>,
 }
 
+impl<H> std::fmt::Debug for ReedSolomon<H> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ReedSolomon").finish()
+    }
+}
+
 impl<H: Hasher> Scheme for ReedSolomon<H> {
     type Commitment = H::Digest;
 
     type Shard = Chunk<H>;
     type ReShard = Chunk<H>;
-
-    type Proof = ();
+    type CheckedShard = Chunk<H>;
```

Comment:
Do we need to wrap `Chunk<H>` in something to get the guarantees we want from this new type? Can you put `Chunk<H>` directly into post-check functions?